lightning-invoice/src/utils.rs

```diff
@@ -67,39 +67,30 @@ pub fn create_phantom_invoice<Signer: Sign, K: Deref>(
 	}
 
 	for hint in phantom_route_hints {
-		for channel in &hint.channels {
-			let short_channel_id = match channel.get_inbound_payment_scid() {
-				Some(id) => id,
-				None => continue,
-			};
-			let forwarding_info = match &channel.counterparty.forwarding_info {
-				Some(info) => info.clone(),
-				None => continue,
-			};
-			invoice = invoice.private_route(RouteHint(vec![
-					RouteHintHop {
-						src_node_id: channel.counterparty.node_id,
-						short_channel_id,
-						fees: RoutingFees {
-							base_msat: forwarding_info.fee_base_msat,
-							proportional_millionths: forwarding_info.fee_proportional_millionths,
-						},
-						cltv_expiry_delta: forwarding_info.cltv_expiry_delta,
-						htlc_minimum_msat: None,
-						htlc_maximum_msat: None,
-					},
-					RouteHintHop {
-						src_node_id: hint.real_node_pubkey,
-						short_channel_id: hint.phantom_scid,
-						fees: RoutingFees {
-							base_msat: 0,
-							proportional_millionths: 0,
-						},
-						cltv_expiry_delta: MIN_CLTV_EXPIRY_DELTA,
-						htlc_minimum_msat: None,
-						htlc_maximum_msat: None,
-					}])
-			);
+		let mut route_hints = filter_channels(
+			hint.channels.clone(),
```

Comment:
Since the function takes hints by value, you can avoid this `clone` by using the de-structuring syntax in the for loop:

```rust
for PhantomRouteHints { channels, phantom_scid, real_node_pubkey } in phantom_route_hints {
```

Reply:
Ah that's neat! Thanks for letting me know